File: Patterns/4. Right Triangle Patterns (Top-Right)/34. Right triangle TR Decreasing Numbers/trdn.py
n = int(input("Enter a Number : "))

for i in range(1,n+1):
    # print spaces
    for j in range(1,n-i+1):
        print(" ",end = " ")
    # print stars
    for j in range(n,n-i,-1):
        print(j, end = " ")
    print()

'''

Output:

        5 
      5 4 
    5 4 3 
  5 4 3 2 
5 4 3 2 1  
  
'''

File: Patterns/4. Right Triangle Patterns (Top-Right)/34. Right triangle TR Decreasing Numbers/trdn.cpp
#include <iostream>
using namespace std;
int main(){
    int n;
    cout<<"Enter a Number : ";
    cin>>n;

    for(int i=1; i<=n; i++){
        // print spaces
        for(int j=1; j<=n-i; j++){
            cout<< "  ";
        }
        // print stars
        for(int j=n; j>=n-i+1; j--){
            cout<<j<<" ";
        }
        cout << '\n';
    }
}

/* 

Output

        5 
      5 4 
    5 4 3 
  5 4 3 2 
5 4 3 2 1  

*/ 